packages/web/src/pages/chat-page/components/ChatMessageTrack.tsx

```diff
@@ -0,0 +1,113 @@
+import { useCallback } from 'react'
+
+import {
+  Kind,
+  Status,
+  makeUid,
+  Track,
+  queueSelectors,
+  playerSelectors,
+  Name,
+  PlaybackSource,
+  queueActions,
+  QueueSource
+} from '@audius/common'
+import { useDispatch, useSelector } from 'react-redux'
+
+import { make } from 'common/store/analytics/actions'
+import MobileTrackTile from 'components/track/mobile/ConnectedTrackTile'
+
+const { makeGetCurrent } = queueSelectors
+const { getPlaying } = playerSelectors
+const { add, play, pause } = queueActions
+
+type ChatMessageTrackProps = {
+  track: Track | undefined | null
+  status: Status
+  errorMessage: string | null | undefined
+}
+
+export const ChatMessageTrack = ({
+  track,
+  status,
+  errorMessage
+}: ChatMessageTrackProps) => {
+  const dispatch = useDispatch()
+  const currentQueueItem = useSelector(makeGetCurrent())
+  const uid = track ? makeUid(Kind.TRACKS, track.track_id) : ''
+  const playing = useSelector(getPlaying)
+  const isTrackPlaying =
+    playing &&
+    !!track &&
+    !!currentQueueItem.track &&
+    currentQueueItem.track.track_id === track.track_id
+
+  const recordAnalytics = useCallback(
+    ({ name, source }: { name: Name; source: PlaybackSource }) => {
+      if (!track) return
+      dispatch(
+        make(name, {
+          id: `${track.track_id}`,
+          source
+        })
+      )
+    },
+    [dispatch, track]
+  )
+
+  const onTogglePlay = useCallback(() => {
+    if (!track) return
+    if (isTrackPlaying) {
+      dispatch(pause({}))
+      recordAnalytics({
+        name: Name.PLAYBACK_PAUSE,
+        source: PlaybackSource.DM_TRACK
```

Comment:
we're calling everything in code "chat" instead of "dm", I think we should keep that here too and call this `CHAT_TRACK`

Reply:
got it